### PR TITLE
/ and /collections returns 500 if create_indices has not been run

### PR DESCRIPTION
**Pull request by philvarner**
**Related Issue(s):** 

- n/a

**Proposed Changes:**

1. Make /collections return 500 when has not been initialized or can't connect to db

**PR Checklist:**

- [X] I have added my changes to the [CHANGELOG](https://github.com/stac-utils/stac-server/blob/main/CHANGELOG.md) **or** a CHANGELOG entry is not required.

### CODE REVIEW

**Review by gadomski — APPROVED** [review]

File: src/lib/database.js [hunk 1]
```diff
@@ -417,8 +417,10 @@ async function getCollections(page = 1, limit = 100) {
     return response.body.hits.hits.map((r) => (r._source))
   } catch (e) {
     logger.error('Failure getting collections, maybe none exist?', e)
+    return new Error('Collections not found. This is likely '
+    + 'because the server has not been initialized with create_indices, '
+    + 'cannot connect to the database, or cannot authenticate to the database.')
```
Comment (MattBialas) [inline src/lib/database.js:422]: Would it be helpful to instruct the user what to do if they see this?

Is it a matter of waiting, or are there steps they need to do to rectify this?

File: src/lib/api.js [hunk 2]
```diff
@@ -976,6 +976,15 @@ const getGlobalAggregations = async (endpoint = '') => {
 }
 
 const getCatalog = async function (txnEnabled, backend, endpoint = '') {
+  const collectionsOrError = await backend.getCollections(1, COLLECTION_LIMIT)
```
Comment (MattBialas) [inline src/lib/api.js:979]: There are many cases where a response might contain an error. I don't know that we _**need**_ the `OrError` part in this name, but it doesn't really hurt either

**Review by MattBialas — APPROVED** [review]
Two little comments but otherwise looks good to me 👍

File: src/lib/database.js (hunk 1 above)
Comment (philvarner) [reply src/lib/database.js:422]: my presumption here was that they would look for "create_indices" in the readme for the instruction they didn't do, or look in the logs for a message about not being able to connect to the database. I don't think there's much more we can give specific direction on here.

File: src/lib/api.js (hunk 2 above)
Comment (philvarner) [reply src/lib/api.js:979]: It's named that way because the type is Either[List[Collection],Error], so it's clear that this might be either.

**Pull request merged**